AI/VCAI/Goals/CollectRes.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Goals.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#pragma once
+
+#include "AbstractGoal.h"
+
+struct HeroPtr;
+class VCAI;
+class FuzzyHelper;
+
+namespace Goals
+{
+	class DLL_EXPORT CollectRes : public CGoal<CollectRes>
+	{
+	public:
+		CollectRes()
+			: CGoal(Goals::COLLECT_RES)
+		{
+		}
+		CollectRes(int rid, int val)
+			: CGoal(Goals::COLLECT_RES)
+		{
+			resID = rid;
+			value = val;
+			priority = 2;
```

Comment:
It should be on common header with all that priority constants.

Reply:
Can put them in abstractgoal.h or in fuzzy helper (it is actually responsible for priority). technically they should not be hardcoded